cli/cli/main.go

```diff
@@ -40,8 +46,51 @@ func main() {
 	})
 
 	if err := commands.RootCmd.Execute(); err != nil {
-		// We don't actually need to print the error because Cobra will do it for us
+		maybeCleanedError := getErrorMessageToBeDisplayedOnCli(err)
+		// cobra uses this method underneath as well to print errors
+		// so just used it directly.
+		commands.RootCmd.PrintErrln("Error: ", maybeCleanedError.Error())
 		os.Exit(errorExitCode)
 	}
 	os.Exit(successExitCode)
 }
+
+func getErrorMessageToBeDisplayedOnCli(errorWithStacktrace error) error {
+	// if we are running in the debug mode, just return the error with stack-traces back to the client
+	if logrus.GetLevel() == logrus.DebugLevel {
+		return errorWithStacktrace
+	}
+
+	// silently catch the file logger error and print it in the debug mode
+	// users should not worry about this error
+	// downside is that we may lose stack-traces during file logger failures
+	fileLogger, err := out.GetFileLogger()
```

Comment:
now with a second review on this, what do you think about moving this logic inside the `filelogger` so you can add and expose only the `out.Errorln` and `out.Infof` methods (both do not return anything) and the error handling will be centralized an encapsulated inside the `out` package. Because now you are instantiating the object but not injecting it in any place, it's not necessary to respect the interface so far. But this is just a suggestion, I'm ok if you prefer to leave it as it is.

Reply:
I just like this better - because it's very verbose i.e `FileLogger` ( is logging only to the file). I can take a look at in next pr. What i agree with you is -- i can create separate `file_logger` package -- and have file_logger.InfoF etc.. This way it's both encapsulated and verbose at the same time? @leoporoli

Reply:
makes sense to have a `file_logger` package, I like it